src/main/java/com/internet/shop/web/filter/AuthorizationFilter.java

```diff
@@ -0,0 +1,94 @@
+    private static final Injector injector = Injector.getInstance("com.internet.shop");
+    private final UserService userService = (UserService) injector
+            .getInstance(UserService.class);
+    private Map<String, List<Role.RoleName>> protectedUrls;
+
+    @Override
+    public void init(FilterConfig filterConfig) throws ServletException {
+        protectedUrls = new HashMap<>();
+        protectedUrls.put("/user/all", List.of(Role.RoleName.ADMIN));
+        protectedUrls.put("/user/delete", List.of(Role.RoleName.ADMIN));
+        protectedUrls.put("/admin/products", List.of(Role.RoleName.ADMIN));
+        protectedUrls.put("/admin/main", List.of(Role.RoleName.ADMIN));
+        protectedUrls.put("/admin/orders", List.of(Role.RoleName.ADMIN));
+        protectedUrls.put("/product/add", List.of(Role.RoleName.ADMIN));
+        protectedUrls.put("/product/delete", List.of(Role.RoleName.ADMIN));
+        protectedUrls.put("/order/delete", List.of(Role.RoleName.ADMIN));
+
+        protectedUrls.put("/user/orders", List.of(Role.RoleName.USER));
+        protectedUrls.put("/product/buy", List.of(Role.RoleName.USER));
+        protectedUrls.put("/order/create", List.of(Role.RoleName.USER));
+        protectedUrls.put("/shopping-cart/products", List.of(Role.RoleName.USER));
+        protectedUrls.put("/shopping-cart/delete", List.of(Role.RoleName.USER));
+    }
+
+    @Override
+    public void doFilter(ServletRequest request, ServletResponse response,
+                         FilterChain chain) throws IOException, ServletException {
+        HttpServletRequest req = (HttpServletRequest) request;
+        HttpServletResponse resp = (HttpServletResponse) response;
+
+        String requestUrl = req.getServletPath();
+
+        if (protectedUrls.get(requestUrl) == null) {
+            chain.doFilter(req, resp);
+            return;
+        }
+
+        Long userId = (Long) req.getSession().getAttribute(LoginController.USER_ID);
+        if (userId == null) {
+            resp.sendRedirect(req.getContextPath() + "/user/login");
+            return;
+        }
+        User user;
+        try {
+            user = userService.get(userId);
+        } catch (NoSuchElementException e) {
+            resp.sendRedirect(req.getContextPath() + "/user/registration");
+            return;
+        }
+
+        if (isAuthorized(user, protectedUrls.get(requestUrl))) {
+            chain.doFilter(req, resp);
+        } else {
+            req.getRequestDispatcher("/WEB-INF/view/accessDenied.jsp").forward(req, resp);
+        }
+    }
+
+    @Override
+    public void destroy() {
+    }
+
+    private boolean isAuthorized(User user, List<Role.RoleName> authorizedRoles) {
+        return authorizedRoles.stream()
+                .anyMatch(r -> user.getRoles().stream()
+                        .map(Role::getRoleName)
+                        .collect(Collectors.toList())
+                        .contains(r));
```

Comment:
we will create a stream and then a list for each role in `authorizedRoles`
let's try to simplify this

Reply:
Done